Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -68,14 +68,18 @@ class MessageDTO: NSManagedObject {
     
     override func willSave() {
         super.willSave()
+
+        guard !isDeleted else {
+            return
+        }
         
         // Manually mark the channel as dirty to trigger the entity update and give the UI a chance
         // to reload the channel cell to reflect the updated preview.
-        if let channel = previewOfChannel, !channel.hasChanges, isUpdated {
+        if let channel = previewOfChannel, !channel.hasChanges, !channel.isDeleted, channel.lastMessageAt == createdAt {
```

Comment:
Can you please elaborate on the last check `channel.lastMessageAt == createdAt`?
 
Updates should be triggered when there are changes to other messages too (e.g. reactions), but I might be lacking some context here.

Reply:
This part is to change the preview message of the channel (In the channel list), so the preview message that we will be analysing is always the last message of the channel. To limit the number of times this part of the codebase is executed, we make sure that the channel is not deleted and that is in fact the last message.

Reply:
got it, thanks for the explanation 👍